tests/common/defender/aws_iot_tests_defender_api.c

```diff
@@ -130,10 +133,13 @@ TEST_TEAR_DOWN( Full_DEFENDER )
 {
     AwsIotDefender_Stop();
 
-    if( _reportAccepted || _reportRejected )
+    /* Actually get defender callback. */
+    if( _callbackInfo.eventType != -1 )
```

Comment:
Chatted offline. It seems in this implementation, we leave throttling handling to user. To be specific, application is increasing _DEFENDER_PUBLISH_INTERVAL_SECONDS, and parsing _callbackInfo.eventType to determine what to do next. 

This may not be the best solution. Since:
1. This creates unreliable test. Throttling is usually per region per account, a fix number with single attempt would never reliably work. 
2. Inconsistent with other AWS clients. Where usually in other clients, users can input retry scheme, and max retry. (e.g. typical retry scheme -- exponential backoff delay. typical max retry -- 3.)


This review comment is more about design instead of this specific PR. Apologies.

Reply:
I agree with #1. And I don't have a good solution now.

#2 will not be a problem for actual user. Defender is a daemon application that will send metrics as slow as 5 minutes (unless user keeps start, stop, start, stop...)

So this problem is scoped to our testing.